Del 2/main.py

```diff
@@ -124,30 +128,95 @@ def turtle():
     print("All paths completed.")
 
 
+def profile(file_name="Mediamodifier-Design"):
+    """
+    Used to profile the parse_svg function on an svg file using cProfile and snakeviz.
+
+    The file can be manually specified.
+    """
+    import cProfile  # pylint: disable=import-outside-toplevel
+    import os  # pylint: disable=import-outside-toplevel
+
+    profiler = cProfile.Profile()
+    profiler.runcall(parse_svg, "app/svg/sample_svgs/{}.svg".format(file_name))
+    profiler.dump_stats("latest.log")
+
+    os.system('"{}" -m snakeviz latest.log'.format(sys.executable))
+
+
+def test():
+    """
+    Used to run the tests
+
+    Returns an exit code
+    """
+    import pytest  # pylint: disable=import-outside-toplevel
+
+    return pytest.main(["-x", "tests"])
+
+
+def help(func_name=None):  # pylint: disable=redefined-builtin
+    """
+    Used to relay information about the different CLI commands
+    to the user.
+    """
+    funcs = [plot, turtle, profile, test, help]
+    if func_name is None:
+        for func in funcs:
+            print("\n{}: {}".format(func.__name__, func.__doc__))
+        return
+
+    for func in funcs:
+        if func_name == func.__name__:
+            print("\n{}: {}".format(func.__name__, func.__doc__))
+            return
+
+    print("The command {} does not exist".format(func_name))
+
+
 if __name__ == "__main__":
     if len(sys.argv) > 1 and sys.argv[1] == "test":
-        import pytest  # pylint: disable=import-outside-toplevel
-
-        sys.exit(pytest.main(["-x", "tests"]))
+        sys.exit(test())
 
     if len(sys.argv) > 1 and sys.argv[1] == "plot":
-        plot()
-        sys.exit()
+        try:
+            if len(sys.argv) > 2:
+                plot(file_names=sys.argv[2:])
+            else:
+                plot()
+        except FileNotFoundError as e:
+            print(e)
+        finally:
+            sys.exit()
 
     if len(sys.argv) > 1 and sys.argv[1] == "profile":
-        import cProfile  # pylint: disable=import-outside-toplevel
-        import os  # pylint: disable=import-outside-toplevel
-
-        profile = cProfile.Profile()
-        profile.runcall(parse_svg, "app/svg/sample_svgs/Mediamodifier-Design.svg")
-        profile.dump_stats("latest.log")
-
-        os.system("{} -m snakeviz latest.log".format(sys.executable))
-        sys.exit()
+        try:
```

Comment:
Why the try / except når exception-handlingen bare printer ut erroren også exiter?

Reply:
Da slipper brukeren å bli redd for en lang error trace slikt at de tror de gjorde noe feil eller ødela noe. Denne måten printer bare selve feilen